### PR TITLE
Table Wrapper changes

### PR DESCRIPTION
**Pull request by Malayvarma27**
Test link: http://10.249.253.4:3000
token: eyJhbGciOiJSUzI1NiIsImtpZCI6IlloNmpYd3RRcUMwQTlVUmRac005ZlFLS0k3VlI5S2Jvc3BrRUdScVRjQ0EifQ.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJzdmMtb2JqZWN0c2NhbGUtZG9tYWluLWMyMCIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VjcmV0Lm5hbWUiOiJvYmplY3RzY2FsZS1vcGVyYXRvci10b2tlbi1wa2h4MiIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJvYmplY3RzY2FsZS1vcGVyYXRvciIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6IjZiYmNmNDk3LTgzOWYtNDZjMC1hMzQ5LTg3ZTkzY2UwN2Y5NCIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDpzdmMtb2JqZWN0c2NhbGUtZG9tYWluLWMyMDpvYmplY3RzY2FsZS1vcGVyYXRvciJ9.MzDyDe3E8WjsW8dyz9Y2bNEKA3VgS7DPi4ScjUgr0FMHGuLekUMHS_4PGDJV52nuF1YOzD-lUrlQ7n-BXsY532Gd1WVK9cg1ykfj7Zlfqqm4Sp9DVUIH9CDB3E9B1V-iMstEgytmH1uvoOuBn1RPzLUmHacOu8VjkG05IEZPtYyEyPIu8jSi5euEqJ-rfBE-kTPAwCExYVkmvmGWg5f2dBdqLInY60jjXLpliFocELkXft_GhXzu3mGbMSaWQ2qhOwYtn-SjppVJNzfgQRmMyDPhPvIFaVNIZy9Mn_MAVd7XWkfP7r4oBl3VsaptB8vWSx7W5_g-NulvZU8vxAMEAg

TableWrapper

### CODE REVIEW

File: src/components/table/Table.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+                         <td className="left">Only if you have no cookies</td>
+                     </tr>
+                 </tbody>
+             </Table>
+         </div>
+     ))
+     .add("Vertical Tables", () => (
+         <div style={{width: "50%", paddingLeft: "20px"}}>
+             <Table isVertical={true}>
+                 <tbody>
```
Comment (akate1) [inline src/components/table/Table.stories.tsx:258]: some stories are suing  `<thead> `and some are not. any reason ? what is recommended/standard way to render table ?

File: src/components/table/Table.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+                             <th className="left">Weakness</th>
+                         </tr>
+                     </thead>
+                     <tbody>
+                         <tr>
+                             <td className="left">Frankenstein</td>
+                             <td className="left">Fire</td>
+                         </tr>
+                         <tr>
+                             <td className="left">Dracula</td>
```
Comment (akate1) [inline src/components/table/Table.stories.tsx:421]: Do we also have class for `right `align items ?

Comment (Malayvarma27) [reply src/components/table/Table.stories.tsx:421]: No we do not have a class for right

File: src/components/table/Table.stories.tsx (hunk 1 above)
Comment (Malayvarma27) [reply src/components/table/Table.stories.tsx:258]: <thead> is not there for vertical tables only. So for the vertical table our structure is like <tr><th /><td /></tr>. We are creating a row in which first column is the header and rest columns contains data.

**Review by akate1 — CHANGES_REQUESTED** [review]

File: src/styles/components/Table.scss [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                        th, td {
+                            border-top: $border-default solid $gray-300;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    &.table-noborder {
```
Comment (akate1) [inline src/styles/components/Table.scss:95]: table-noborder and table-vertical  classes are repeated. is there any way to combine it ?

File: src/styles/components/Table.scss [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        & th {
+            &:not(:last-child):after {
+                width: 0;
+                height: 0;
+            }
+
+            &:first-child {
+                padding-left: 1.2rem;
+                background: transparent;
+                border-right: $border-default solid $gray-300;
```
Comment (akate1) [inline src/styles/components/Table.scss:116]: Please create constant for border -right as it is repeating

File: src/styles/components/Table.scss [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        font-weight: $font-weight-regular;
+        line-height: $line-height-20;
+        letter-spacing: 0;
+        padding: .5rem .8rem;
+
+        &:first-child, &.left:first-child {
+            padding-left: 1.2rem;
+        }
+
+        &.objectstore-certificate-label {
```
[the remark this answers is not in the document]
Comment (Malayvarma27) [reply src/styles/components/Table.scss]: done. remove this from wrapper and will make changes in plugin repo.

File: src/styles/components/Table.scss (hunk 3 above)
Comment (Malayvarma27) [reply src/styles/components/Table.scss:72]: actually its not repeated ..its like combination of different classes. I have refactored it and used the compact class in both noborder and vertical class rather that the other way round like it was before

File: src/styles/components/Table.scss (hunk 4 above)
Comment (Malayvarma27) [reply src/styles/components/Table.scss]: done. Since the value is same for all borders, i have created a constant in common border file  ie, table-border

**Review by akate1 — APPROVED** [review]
Tested. LGTM

**Review by ShitalSD — APPROVED** [review]

**Pull request merged**